### ISSUE TITLE
Clip management in the admin

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#2872]
### Share your bug report, feature request, or comment.

- Edit clip metadata (titles, descriptions, etc)
- Delete clips

**Comment by armadi1809** [comment]
@gabek Are you still planning on pursuing this? If so, would like to see what I can do here.

**Comment by gabek** [comment]
@armadi1809 Thanks for checking in! This is actually a _really, really_ large set of multiple features.

Before we can build the admin interface for managing clips, we need all of the functionality built to actually create clips, and have all the functionality settled upon.

I have about half of the functionality in this PR, but there's still so much more to go: https://github.com/owncast/owncast/pull/3395

But before I feel comfortable adding these considerable features to Owncast, I want to pursue a backend refactor to make it easier, and more sustainable, to add this kind of functionality. While not as big of an endeavor as rewriting the Owncast web frontend that happened, but this has the same goals in mind.

I was also hoping to have scheduled events in before replays and clips were in, since I think it makes sense. But again, I don't want to build scheduled events until I give the backend some attention.

So all of that to say, we're really far off from being able to build any kind of UI for replays and clips, and when we are ready, it will take a lot of discussion and planning before it can be worked on.

I hope that's not discouraging!

**Comment by armadi1809** [comment]
No, that's fair. I guess I will keep an eye on the PR you listed and on the repo in general and try to contribute to this effort whenever it's appropriate. Thank you for the detailed response!

**Comment by gabek** [comment]
If you're curious of what I kind of saw as our tentative roadmap, I have it at https://owncast.online/roadmap/. But everything is subject to change :)

<blockquote><img src="/images/owncast-logo-1000x1000.png" width="48" align="right"><div><img src="https://owncast.online/favicon.png" height="14"> Owncast</div><div><strong><a href="/roadmap/">Owncast public roadmap</a></strong></div><div>While nothing is set in stone, Owncast tries to set a relatively clear roadmap for future work and features. This enables the project to discuss things far in advance to make the best possible decisions as well as allowing Owncast users to know what types of updates they can expect.</div></blockquote>
